cni/cni.go

```diff
@@ -26,8 +26,8 @@ const (
 	Internal = "internal"
 )
 
-// Supported CNI versions.
-var VersionsSupported = []string{"0.2.0", "0.3.0"}
+// Oldest Supported CNI versions
+var OldestVersionSupported = "0.2.0"
```

Comment:
@Tbbie could we revert this to explicitly indicating supported versions? The current release versioning seems to correspond with greatest supported version. We can pull this change into a separate PR and discuss it.

Reply:
Just to be clear, you're requesting I open-code this list to the current return value of `cniVers.VersionsStartingFrom(OldestVersionSupported)`?

Reply:
That was the idea but seems counter productive. @Debjit would we be okay with using the [cni ](github.com/containernetworking/cni) import versioning to limit the max supported version? looking at the [releases ](https://github.com/containernetworking/cni/releases) it doesn't seem to be revved that frequently

Reply:
So I was poking around the CNI spec a little bit, and it's possible this change is faulty, as if I'm skimming it correctly, [by declaring we support config files 0.4.0 or newer, we are also opening ourselves up to having `CHECK` called for such config files](https://github.com/containernetworking/cni/blob/v1.1.2/pkg/skel/skel.go#L220-L241), but [we currently pass `nil` as the check callback](https://github.com/microsoft/windows-container-networking/blob/master/cni/plugin.go#L47-L48). A quick look at the CNI skeleton code [doesn't show any checks for `nil` in the relevant code](https://github.com/containernetworking/cni/blob/v1.1.2/pkg/skel/skel.go#L166-L172), so this would panic if we hit it. (`CHECK` was introduced in CNI spec 0.4.0 and it also changed the `DEL` API; 0.4.0 to 1.0.0 looks like it only changed config and structures, according to [the "major changes" list](https://github.com/containernetworking/cni/blob/main/SPEC.md#released-versions))

A quick look at BuildKit (my test case) doesn't show any calls to the go-cni `Check` API which wraps this, so I haven't exercised this change in my limited testing.

I am not totally on top of how CNI works, so when I was building this PR I was mostly looking at the config-file parser, but perhaps we need to (for purposes of this PR) keep the existing `VersionsSupported` array, and push the actual supported versions change out to require an implementation of `CHECK` as well. (And probably a more-careful check of changes in the spec)

If I'm right and the version bump was incorrect, my understanding is we can still use the latest library (we were already on library 0.8.1 which appears to have supported spec 0.4.0), so we only need to revert our version declaration to avoid the behaviour change. We can also separately go ahead and upgrade the internal structures to types/100 or later in the meantime, as again, we are already using version types/040 with spec 0.3.0 support. (Again, might be worth validating in the spec that the meaning of those fields hasn't changed, in case something changed that isn't caught by simple Go compiler type-checking.)

Reply:
CHECK is not implemented in containerd or cri-o. I added the check code in go-cni however discontinued it as it involves a significant amount of work to the runtime/cri-api/kubelet.

Reply:
So maybe just a function that returns a "not implemented" error would be sufficient to avoid the panic, in case that rework is done in the future, or another CNI user appears.

Reply:
I see the [cni plugins](https://github.com/containernetworking/plugins/blob/abee8ccc0de94610e5d14488479bcb87ad64776c/plugins/main/windows/win-bridge/win-bridge_windows.go#L212C1-L215C2) callback function returns `nil` which emulates success? `not implemented` would allow us to catch unknown or future scenarios earlier though. @MikeZappa87 are we ok with adding a no-op check command callback + an issue to track the work to implement it?

Reply:
I think we can proceed with returning "not implemented" for CHECK, and adding both 0.4.0 and 1.0.0 as the supported versions in the array. We can track the remaining work with the below issues based on requirement to support them (doesn't seem to be widely used at the moment):
1. Implement CHECK; update config examples to include disableCheck field (0.4.0) 
2. Remove non-list configurations; remove version field of interfaces array (1.0.0)